moebot_bot/bot/commands/channelRotationScheduler.go

```diff
@@ -0,0 +1,206 @@
+package commands
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/camd67/moebot/moebot_bot/util"
+	"github.com/camd67/moebot/moebot_bot/util/db"
+	"github.com/camd67/moebot/moebot_bot/util/moeDiscord"
+)
+
+type ChannelRotationScheduler struct {
+	schedulerType db.SchedulerType
+	session       *discordgo.Session
+}
+
+func NewChannelRotationScheduler(schedulerType db.SchedulerType, session *discordgo.Session) *ChannelRotationScheduler {
+	return &ChannelRotationScheduler{schedulerType, session}
+}
+
+func (s *ChannelRotationScheduler) Execute(operationId int64) {
+	channelRotation, err := db.ChannelRotationQuery(operationId)
```

Comment:
This function warrants at least a block comment describing the general flow of what's going on here